src/connection-pool-party.js

```diff
@@ -484,4 +516,86 @@ export default class ConnectionPoolParty extends EventEmitter {
         err => err,
       );
   }
+
+  _prioritizePools = () => {
+    if (!this.warmedUp) {
+      debug('_prioritizePools called before warmup completed. this should not happen.');
+      return;
+    }
+    if (this._healingPromise) {
+      debug('_prioritizePools called during heal, skipping this run');
+      return;
+    }
+    if (this.pools.length === 0) {
+      debug('_prioritizePools called when no pools exist, this should not happen');
+      return;
+    }
+    if (this.pools.length === 1) {
+      debug('_prioritizePools called when only one pool exists, skipping this run');
+      return;
+    }
+    if (this._prioritizePromise) {
+      debug('_prioritizePools called again while already in progress');
+    } else {
+      debug('_prioritizePools called');
+    }
+    const firstPoolPriority = this.pools[0].dsn.priority;
+    if (firstPoolPriority === undefined || firstPoolPriority === 0) {
+      debug('first pool has top priority, no need to prioritize');
+      return;
+    }
+    const higherPriorityPools = this.pools.filter(
+      pool => pool.dsn.priority < firstPoolPriority,
+    );
+    if (higherPriorityPools.length === 0) {
+      debug('unexpected priority config on DSNs, unable to prioritize');
+      return;
+    }
+    const unhealthyPriorityPools = higherPriorityPools.filter(
+      pool => !pool.connection.connecting && !pool.connection.connected,
+    );
+    this._prioritizePromise = this._prioritizePromise || Promise.resolve(unhealthyPriorityPools)
+      .then((unhealthyPools) => {
+        // If all the pools of a higher priority than index 0 are healthy, we can
+        // skip the heal. If there are any unhealthy pools,
+        // we need to heal them before sorting.
+        if (unhealthyPools.length === 0) {
+          this.pools.sort(poolPrioritySort);
```

Comment:
We don't need to sort here, it will be sorted in the following `.then`.

Reply:
i believe that `_healPools` will return false if there are no unhealthy pools passed to it, which would result in an early return without a sort in the following `then`, so we do need it here

Reply:
Well, we are returning `true` in the next two line; which will set `anyHealthyPools` to `true` in the following `then`.

Reply:
oops, you're right, i think i got part way through refactoring to a single sort and forgot to take this part out

Reply:
oops, you're right, i think i got part way through refactoring to a single sort and forgot to take this part out